[CI] ES94BloomFilterDocValuesFormatTests testMergedBloomFilterUsesMaxSize failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #12326 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/ggtz3ym6bu7b4)
- [elasticsearch-periodic-platform-support #12326 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/aj446mnveyqvu)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.codec.bloomfilter.ES94BloomFilterDocValuesFormatTests.testMergedBloomFilterUsesMaxSize" -Dtests.seed=AEA9DB71835F47C1 -Dtests.locale=en-LR -Dtests.timezone=ROK -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.codec.bloomfilter.ES94BloomFilterDocValuesFormatTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testMergedBloomFilterUsesMaxSize'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: is a value greater than <1>
     but: <1> was equal to <1>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testMergedBloomFilterUsesMaxSize
- [main] 2 consecutive failures in step rocky-8_platform-support-unix
- [main] 1 consecutive failures in pipeline elasticsearch-periodic-platform-support
- [main] 2 failures in test testMergedBloomFilterUsesMaxSize (0.7% fail rate in 288 executions)
- [main] 2 failures in step rocky-8_platform-support-unix (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -593,9 +593,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.action.ExternalNdJsonMultiScanPushdownIT
   method: testMultiScanColdDoesNotDoubleCountWarmCountStar
   issue: https://github.com/elastic/elasticsearch/issues/150837
-- class: org.elasticsearch.index.codec.bloomfilter.ES94BloomFilterDocValuesFormatTests
-  method: testMergedBloomFilterUsesMaxSize
-  issue: https://github.com/elastic/elasticsearch/issues/150843
 - class: org.elasticsearch.xpack.esql.qa.ndjson.NdJsonFormatSpecIT
   method: test {csv-spec:external-basic.forkTwoBranches [AZURE]}
   issue: https://github.com/elastic/elasticsearch/issues/150844

--- a/server/src/test/java/org/elasticsearch/index/codec/bloomfilter/ES94BloomFilterDocValuesFormatTests.java
+++ b/server/src/test/java/org/elasticsearch/index/codec/bloomfilter/ES94BloomFilterDocValuesFormatTests.java
@@ -23,6 +23,7 @@ import org.apache.lucene.index.IndexWriter;
 import org.apache.lucene.index.IndexWriterConfig;
 import org.apache.lucene.index.LeafReader;
 import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.LogMergePolicy;
 import org.apache.lucene.index.StandardDirectoryReader;
 import org.apache.lucene.tests.analysis.MockAnalyzer;
 import org.apache.lucene.tests.codecs.asserting.AssertingCodec;
@@ -214,7 +215,9 @@ public class ES94BloomFilterDocValuesFormatTests extends ESTestCase {
                     return (flushCount.getAndIncrement() % 2 == 0) ? smallSize : largeSize;
                 }
             }));
-            conf.setMergePolicy(newLogMergePolicy());
+            LogMergePolicy mergePolicy = newLogMergePolicy();
+            mergePolicy.setMergeFactor(1000);
+            conf.setMergePolicy(mergePolicy);
             conf.setMaxBufferedDocs(10);
             conf.setUseCompoundFile(false);
             try (IndexWriter writer = new IndexWriter(directory, conf)) {

